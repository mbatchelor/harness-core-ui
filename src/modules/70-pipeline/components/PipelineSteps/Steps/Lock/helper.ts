/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { MultiTypeInputType } from '@wings-software/uicore'
import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'
import type { StepElementConfig } from 'services/cd-ng'
import type { StringsMap } from 'stringTypes'

export const SCOPE_KEYS = {
  PIPELINE: 'PIPELINE',
  STAGE: 'STAGE',
  STEP_GROUP: 'STEP_GROUP',
  PLAN: 'PLAN'
}

export interface LockData extends StepElementConfig {
  spec: {
    key: string
    scope: string
  }
}

export interface LockProps {
  initialValues: LockData
  onUpdate?: (data: LockData) => void
  stepViewType: StepViewType
  allowableTypes: MultiTypeInputType[]
  isNewStep?: boolean
  inputSetData?: {
    template?: LockData
    path?: string
    readonly?: boolean
  }
  onChange?: (data: LockData) => void
  readonly?: boolean
}

export const getScopeOptions = (
  getString: (key: keyof StringsMap, vars?: Record<string, any> | undefined) => string
) => [
  { label: getString('common.pipeline'), value: SCOPE_KEYS.PIPELINE },
  { label: getString('common.stage'), value: SCOPE_KEYS.STAGE },
  { label: getString('stepGroup'), value: SCOPE_KEYS.STEP_GROUP },
  { label: getString('common.subscriptions.overview.plan'), value: SCOPE_KEYS.PLAN }
]
