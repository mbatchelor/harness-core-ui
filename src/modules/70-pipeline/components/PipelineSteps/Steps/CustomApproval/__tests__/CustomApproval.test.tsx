/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
 */

import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react'
import { StepType } from '@pipeline/components/PipelineSteps/PipelineStepInterface'
import { StepFormikRef, StepViewType } from '@pipeline/components/AbstractSteps/Step'
import { getDefaultCriterias } from '@pipeline/components/PipelineSteps/Steps/JiraApproval/helper'
import { TestStepWidget, factory } from '@pipeline/components/PipelineSteps/Steps/__tests__/StepTestUtil'
import { CustomApproval } from '../CustomApproval'
import {
  getCustomApprovalDeploymentModeProps,
  getCustomApprovalEditModeProps,
  getCustomApprovalInputVariableModeProps
} from './CustomApprovalTestHelper'

jest.mock('@common/components/YAMLBuilder/YamlBuilder')

describe('CustomApproval tests', () => {
  beforeEach(() => {
    factory.registerStep(new CustomApproval())
  })

  test('Basic snapshot - inputset mode', async () => {
    const props = getCustomApprovalDeploymentModeProps()
    const { container, getByText, queryByText } = render(
      <TestStepWidget
        initialValues={props.initialValues}
        type={StepType.CustomApproval}
        stepViewType={StepViewType.InputSet}
      />
    )

    fireEvent.click(getByText('Submit'))
    await waitFor(() => queryByText('Errors'))
    expect(container).toMatchSnapshot('input set with errors')
  })

  test('Basic snapshot - deploymentform mode', async () => {
    const props = getCustomApprovalDeploymentModeProps()
    const { container } = render(
      <TestStepWidget
        initialValues={props.initialValues}
        type={StepType.CustomApproval}
        stepViewType={StepViewType.DeploymentForm}
      />
    )

    expect(container).toMatchSnapshot('custom-approval-deploymentform')
  })

  test('deploymentform mode - readonly', async () => {
    const props = getCustomApprovalDeploymentModeProps()
    const { container } = render(
      <TestStepWidget
        initialValues={props.initialValues}
        type={StepType.CustomApproval}
        stepViewType={StepViewType.DeploymentForm}
        inputSetData={{ path: '', readonly: true }}
      />
    )

    expect(container).toMatchSnapshot('custom-approval-deploymentform readonly')
  })

  test('Basic snapshot - inputset mode but no runtime values', async () => {
    const props = getCustomApprovalDeploymentModeProps()
    const { container } = render(
      <TestStepWidget
        initialValues={props.initialValues}
        type={StepType.CustomApproval}
        template={{ spec: { approvalCriteria: getDefaultCriterias(), rejectionCriteria: getDefaultCriterias() } }}
        stepViewType={StepViewType.InputSet}
      />
    )
    expect(container).toMatchSnapshot('custom-approval-inputset-noruntime')
  })

  test('Basic snapshot - input variable view', () => {
    const props = getCustomApprovalInputVariableModeProps()
    const { container } = render(
      <TestStepWidget
        initialValues={props.initialValues}
        type={StepType.CustomApproval}
        template={{ spec: { approvalCriteria: getDefaultCriterias(), rejectionCriteria: getDefaultCriterias() } }}
        stepViewType={StepViewType.InputVariable}
        customStepProps={props.customStepProps}
      />
    )

    expect(container).toMatchSnapshot('custom-approval-input variable view')
  })

  test('Edit stage - readonly', async () => {
    const ref = React.createRef<StepFormikRef<unknown>>()
    const props = getCustomApprovalEditModeProps()
    const { container } = render(
      <TestStepWidget
        initialValues={props.initialValues}
        type={StepType.CustomApproval}
        stepViewType={StepViewType.Edit}
        ref={ref}
        readonly={true}
      />
    )

    expect(container).toMatchSnapshot('edit stage view readonly')
  })

  test('Minimum time cannot be less than 10s', () => {
    const response = new CustomApproval().validateInputSet({
      data: {
        name: 'Test A',
        identifier: 'Test A',
        timeout: '1s',
        type: 'CustomApproval',
        spec: {
          approvalCriteria: getDefaultCriterias(),
          rejectionCriteria: getDefaultCriterias()
        }
      },
      template: {
        name: 'Test A',
        identifier: 'Test A',
        timeout: '<+input>',
        type: 'CustomApproval',
        spec: {
          approvalCriteria: getDefaultCriterias(),
          rejectionCriteria: getDefaultCriterias()
        }
      },
      viewType: StepViewType.TriggerForm
    })
    expect(response).toMatchSnapshot('Value must be greater than or equal to "10s"')
  })
})
