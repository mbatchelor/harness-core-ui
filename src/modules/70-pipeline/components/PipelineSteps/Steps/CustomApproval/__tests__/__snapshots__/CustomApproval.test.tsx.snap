// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomApproval tests Basic snapshot - deploymentform mode: custom-approval-deploymentform 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main"
    >
      <div />
    </form>
    <p
      class="StyledProps--font StyledProps--main"
    >
      Errors
    </p>
    <pre>
      {}
    </pre>
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Submit
      </span>
    </button>
  </div>
</div>
`;

exports[`CustomApproval tests Basic snapshot - input variable view: custom-approval-input variable view 1`] = `
<div>
  <div
    class="variablesListTable variablePaddingL3"
  />
</div>
`;

exports[`CustomApproval tests Basic snapshot - inputset mode but no runtime values: custom-approval-inputset-noruntime 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main"
    >
      <div />
    </form>
    <p
      class="StyledProps--font StyledProps--main"
    >
      Errors
    </p>
    <pre>
      {}
    </pre>
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Submit
      </span>
    </button>
  </div>
</div>
`;

exports[`CustomApproval tests Basic snapshot - inputset mode: input set with errors 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main"
    >
      <div />
    </form>
    <p
      class="StyledProps--font StyledProps--main"
    >
      Errors
    </p>
    <pre>
      {}
    </pre>
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Submit
      </span>
    </button>
  </div>
</div>
`;

exports[`CustomApproval tests Edit stage - readonly: edit stage view readonly 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <div
      class="StyledProps--font StyledProps--main customApprovalContainer"
    >
      <div
        class="formGroup lg"
      >
        <div
          class="InputWithIdentifier--txtNameContainer TextInput--main"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
          >
            <div
              class="Text--withDocsTooltip"
              data-tooltip-id="universalInputNameWithId"
            >
              <p
                class="StyledProps--font StyledProps--main"
              >
                Id
              </p>
            </div>
            <p
              class="StyledProps--font StyledProps--main"
            >
              :
            </p>
            <span
              class="bp3-popover-wrapper"
            >
              <span
                class="bp3-popover-target"
              >
                <div
                  class="bp3-editable-text bp3-disabled bp3-editable-text-placeholder InputWithIdentifier--idValue"
                >
                  <span
                    class="bp3-editable-text-content"
                    style="height: 1px; line-height: 1px; min-width: 0;"
                  />
                </div>
              </span>
            </span>
          </div>
          <div
            class="bp3-form-group bp3-disabled"
          >
            <label
              class="bp3-label"
              for="name"
            >
              <span
                class="Tooltip--acenter"
                data-tooltip-id="CustomApprovalForm_name"
              >
                name
              </span>
               
              <span
                class="bp3-text-muted"
              />
            </label>
            <div
              class="bp3-form-content"
            >
              <div
                class="bp3-input-group bp3-disabled"
              >
                <input
                  autocomplete="off"
                  class="bp3-input"
                  disabled=""
                  name="name"
                  placeholder="pipeline.stepNamePlaceholder"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="formGroup sm"
      >
        <div
          class="bp3-form-group bp3-disabled duration"
        >
          <label
            class="bp3-label"
            for="timeout"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="CustomApprovalForm_timeout"
            >
              pipelineSteps.timeoutLabel
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="Layout--horizontal StyledProps--main MultiTypeInput--main MultiTypeInput--disabled"
              style="flex-grow: 1;"
            >
              <div
                class="bp3-input-group bp3-disabled bp3-fill"
              >
                <input
                  class="bp3-input"
                  disabled=""
                  name="timeout"
                  placeholder="Enter w/d/h/m/s/ms"
                  type="text"
                  value="5s"
                />
              </div>
              <button
                class="MultiTypeInput--btn MultiTypeInput--FIXED"
                disabled=""
              >
                <span
                  class="bp3-icon StyledProps--main"
                  data-icon="fixed-input"
                >
                  <svg
                    height="12"
                    viewBox="0 0 9 9"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="divider"
      />
      <div
        class="formGroup sm"
      >
        <div
          class="bp3-form-group bp3-disabled"
        >
          <label
            class="bp3-label"
            for="spec.shell"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="CustomApprovalForm_spec.shell"
            >
              scriptType
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-popover-wrapper Select--main bp3-fill"
            >
              <div
                class="bp3-popover-target"
              >
                <div
                  class="bp3-input-group bp3-disabled"
                >
                  <input
                    autocomplete="off"
                    class="bp3-input"
                    disabled=""
                    name="spec.shell"
                    placeholder="- scriptType -"
                    style="padding-right: 0px;"
                    type="text"
                    value="Bash"
                  />
                  <span
                    class="bp3-input-action"
                  >
                    <span
                      class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                      icon="chevron-down"
                    >
                      <svg
                        data-icon="chevron-down"
                        height="14"
                        viewBox="0 0 16 16"
                        width="14"
                      >
                        <desc>
                          chevron-down
                        </desc>
                        <path
                          d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="formGroup"
      >
        <div
          class="bp3-form-group bp3-disabled"
        >
          <label
            class="bp3-label"
            for="spec.source.spec.script"
          >
            <div
              class="formLabel"
            >
              <span
                class="Tooltip--acenter"
                data-tooltip-id="CustomApprovalForm_spec.source.spec.script"
              >
                script
              </span>
            </div>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="monacoWrapper"
            >
              <textarea
                name="spec.source.spec.script"
              />
              <button
                class="bp3-button bp3-small Button--button StyledProps--font StyledProps--main expandBtn Button--with-current-color Button--iconOnly Button--withLeftIcon"
                type="button"
              >
                <span
                  class="bp3-icon bp3-icon-fullscreen StyledProps--main"
                  icon="fullscreen"
                >
                  <svg
                    data-icon="fullscreen"
                    height="10"
                    viewBox="0 0 16 16"
                    width="10"
                  >
                    <desc>
                      fullscreen
                    </desc>
                    <path
                      d="M3.41 2H5c.55 0 1-.45 1-1s-.45-1-1-1H1C.45 0 0 .45 0 1v4c0 .55.45 1 1 1s1-.45 1-1V3.41L5.29 6.7c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L3.41 2zM6 9c-.28 0-.53.11-.71.29L2 12.59V11c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .55.45 1 1 1h4c.55 0 1-.45 1-1s-.45-1-1-1H3.41l3.29-3.29c.19-.18.3-.43.3-.71 0-.55-.45-1-1-1zm9 1c-.55 0-1 .45-1 1v1.59L10.71 9.3A.965.965 0 0010 9a1.003 1.003 0 00-.71 1.71l3.3 3.29H11c-.55 0-1 .45-1 1s.45 1 1 1h4c.55 0 1-.45 1-1v-4c0-.55-.45-1-1-1zm0-10h-4c-.55 0-1 .45-1 1s.45 1 1 1h1.59l-3.3 3.29a1.003 1.003 0 001.42 1.42L14 3.41V5c0 .55.45 1 1 1s1-.45 1-1V1c0-.55-.45-1-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="formGroup sm"
      >
        <div
          class="bp3-form-group bp3-disabled duration"
        >
          <label
            class="bp3-label"
            for="spec.retryInterval"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="CustomApprovalForm_spec.retryInterval"
            >
              pipeline.customApprovalStep.retryInterval
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="Layout--horizontal StyledProps--main MultiTypeInput--main MultiTypeInput--disabled"
              style="flex-grow: 1;"
            >
              <div
                class="bp3-input-group bp3-disabled bp3-fill"
              >
                <input
                  class="bp3-input"
                  disabled=""
                  name="spec.retryInterval"
                  placeholder="Enter w/d/h/m/s/ms"
                  type="text"
                  value="20s"
                />
              </div>
              <button
                class="MultiTypeInput--btn MultiTypeInput--FIXED"
                disabled=""
              >
                <span
                  class="bp3-icon StyledProps--main"
                  data-icon="fixed-input"
                >
                  <svg
                    height="12"
                    viewBox="0 0 9 9"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="formGroup sm"
      >
        <div
          class="bp3-form-group bp3-disabled duration"
        >
          <label
            class="bp3-label"
            for="spec.scriptTimeout"
          >
            <span
              class="Tooltip--acenter"
              data-tooltip-id="CustomApprovalForm_spec.scriptTimeout"
            >
              pipeline.customApprovalStep.scriptTimeout
            </span>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="Layout--horizontal StyledProps--main MultiTypeInput--main MultiTypeInput--disabled"
              style="flex-grow: 1;"
            >
              <div
                class="bp3-input-group bp3-disabled bp3-fill"
              >
                <input
                  class="bp3-input"
                  disabled=""
                  name="spec.scriptTimeout"
                  placeholder="Enter w/d/h/m/s/ms"
                  type="text"
                  value="2m"
                />
              </div>
              <button
                class="MultiTypeInput--btn MultiTypeInput--FIXED"
                disabled=""
              >
                <span
                  class="bp3-icon StyledProps--main"
                  data-icon="fixed-input"
                >
                  <svg
                    height="12"
                    viewBox="0 0 9 9"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="formGroup"
      >
        <div
          class="bp3-form-group"
        >
          <label
            class="bp3-label"
            for="spec.outputVariables"
          >
            <div
              class="formLabel"
            >
              <span
                class="Tooltip--acenter"
                data-tooltip-id="CustomApprovalForm_spec.outputVariables"
              >
                pipeline.scriptOutputVariables
              </span>
            </div>
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="panel"
            >
              <div
                class="outputVarHeader"
              >
                <span
                  class="label"
                >
                  Name
                </span>
                <span
                  class="label"
                >
                  Type
                </span>
                <span
                  class="label"
                >
                  Value
                </span>
              </div>
              <button
                class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main addButton Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-link"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="bp3-icon bp3-icon-plus StyledProps--main"
                  icon="plus"
                >
                  <svg
                    data-icon="plus"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      plus
                    </desc>
                    <path
                      d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  class="bp3-button-text"
                >
                  addOutputVar
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="box"
      >
        <div
          class="ng-tooltip-native"
        >
          <div
            class="stepSubSectionHeading"
            data-tooltip-id="approvalapprovalCriteria"
          >
            pipeline.approvalCriteria.approvalCriteria
          </div>
        </div>
        <div
          class="tabs"
        >
          <label
            class="bp3-control bp3-radio bp3-disabled StyledProps--font Radio--radio tab StyledProps--main"
          >
            <input
              checked=""
              disabled=""
              type="radio"
            />
            <span
              class="bp3-control-indicator"
            />
            conditions
          </label>
          <label
            class="bp3-control bp3-radio bp3-disabled StyledProps--font Radio--radio tab StyledProps--main"
          >
            <input
              disabled=""
              type="radio"
            />
            <span
              class="bp3-control-indicator"
            />
            common.jexlExpression
          </label>
        </div>
        <div
          class="conditionalContent"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-xxxlarge StyledProps--main alignConditions"
          >
            <span>
              pipeline.approvalCriteria.match
            </span>
            <label
              class="bp3-control bp3-radio bp3-disabled StyledProps--font Radio--radio StyledProps--main"
            >
              <input
                disabled=""
                type="radio"
              />
              <span
                class="bp3-control-indicator"
              />
              pipeline.approvalCriteria.allConditions
            </label>
            <label
              class="bp3-control bp3-radio bp3-disabled StyledProps--font Radio--radio StyledProps--main"
            >
              <input
                checked=""
                disabled=""
                type="radio"
              />
              <span
                class="bp3-control-indicator"
              />
              pipeline.approvalCriteria.anyCondition
            </label>
          </div>
          <div
            class="formGroup"
          >
            <div
              class="criteriaRow"
            >
              <div
                class="headers"
              >
                <span>
                  pipeline.approvalCriteria.field
                </span>
                <span>
                  common.operator
                </span>
                <span>
                  valueLabel
                </span>
              </div>
              <button
                class="bp3-button bp3-disabled bp3-minimal Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
                data-testid="add-conditions"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="bp3-icon bp3-icon-plus StyledProps--main"
                  icon="plus"
                >
                  <svg
                    data-icon="plus"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      plus
                    </desc>
                    <path
                      d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  class="bp3-button-text"
                >
                  add
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Accordion--accordion accordion"
    >
      <div
        class="Accordion--panel"
        data-open="false"
        data-testid="optional-config-panel"
      >
        <div
          class="Accordion--summary"
          data-testid="optional-config-summary"
        >
          <div
            class="Accordion--label"
          >
            common.optionalConfig
          </div>
          <div
            class="Accordion--chevron"
          />
        </div>
        <div
          class="bp3-collapse Accordion--collapse"
        >
          <div
            aria-hidden="false"
            class="bp3-collapse-body"
            style="transform: translateY(-undefinedpx);"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CustomApproval tests Minimum time cannot be less than 10s: Value must be greater than or equal to "10s" 1`] = `
Object {
  "timeout": "Value must be greater than or equal to \\"10s\\"",
}
`;

exports[`CustomApproval tests deploymentform mode - readonly: custom-approval-deploymentform readonly 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main"
    >
      <div />
    </form>
    <p
      class="StyledProps--font StyledProps--main"
    >
      Errors
    </p>
    <pre>
      {}
    </pre>
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Submit
      </span>
    </button>
  </div>
</div>
`;
